### ISSUE TITLE
Resource response not casting timestampz fields after upgrade to 5.8

### ISSUE THREAD

**Issue by edbizarro** [laravel/framework#27922]
- Laravel Version: v5.8.4
- PHP Version: 7.3.3
- Database Driver & Version: PostgresSQL 11 pdo_psql

### Description:

After upgrade to Laravel v5.8.4, all my timestampz fields on resources are returning a string, instead of an array, this is the desired behavior?

My model has:

```php
/**
     * The storage format of the model's date columns.
     *
     * @var string
     */
    protected $dateFormat = 'Y-m-d H:i:sO';

    protected $casts = [
        'updated_at' => 'datetime:Y-m-d H:i:sO',
        'created_at' => 'datetime:Y-m-d H:i:sO',
    ];

    protected $dates = [
        'updated_at',
        'created_at',
    ];

```
Laravel v5.7.28

```php
"created_at": {
    "date": "2019-03-18 14:45:04.000000",
    "timezone_type": 3,
    "timezone": "America\/Sao_Paulo"
},
"updated_at": {
    "date": "2019-03-18 14:45:04.000000",
    "timezone_type": 3,
    "timezone": "America\/Sao_Paulo"
}
```

Laravel v5.8.4

```php
"created_at": "2019-03-18T17:45:04.000000Z",
"updated_at": "2019-03-18T17:45:04.000000Z"
```

### Steps To Reproduce:

- Setup your application with PostgreSQL DB engine.
- Create a table in the DB with one column being of type timestampz
- Create a model that has an attribute $casts has a type 'datetime:Y-m-d H:i:sO'
- Create a Resource class with the created model
- Make a request to an endpoint with a Resource::collection() or single() response

**Comment by staudenmeir** [comment]
I get the same result in Laravel 5.7 and 5.8.

Please post the code of a minimal example.

**Comment by edbizarro** [comment]
> I get the same result in Laravel 5.7 and 5.8.
> 
> Please post the code of a minimal example.

Model MediaPlan.php

```php
?php

namespace App;

use App\Events\MediaPlanCreated;
use App\Events\MediaPlanUpdated;
use App\Traits\HasUuid;
use Illuminate\Database\Eloquent\Model;

/** @mixin \Eloquent */
class MediaPlan extends Model
{
    use HasUuid;

    protected $fillable = [
        'name',
        'account_uuid',
        'status',
        'investment',
        'region',
        'target_media',
        'date_start',
        'date_end',
    ];

    /**
     * The storage format of the model's date columns.
     *
     * @var string
     */
    protected $dateFormat = 'Y-m-d H:i:sO';

    protected $casts = [
        'date_start' => 'datetime:Y-m-d H:i:sO',
        'date_end' => 'datetime:Y-m-d H:i:sO',
        'updated_at' => 'datetime:Y-m-d H:i:sO',
        'created_at' => 'datetime:Y-m-d H:i:sO',
    ];

    protected $dates = [
        'date_start',
        'date_end',
        'updated_at',
        'created_at',
    ];
```

MediaPlanResource.php

```php
<?php

namespace App\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;

class MediaPlanResource extends JsonResource
{
    /**
     * Transform the resource collection into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function toArray($request)
    {
        return [
            'uuid' => $this->uuid,
            'name' => $this->name,
            'account_uuid' => $this->account_uuid,
            'account' => new AccountResource($this->whenLoaded('account')),
            'date_start' => $this->date_start,
            'date_end' => $this->date_end,
            'investment' => number_format($this->investment, 2, ',', '.'),
            'region' => $this->region,
            'target_media' => $this->target_media,
            'status' => $this->status,
            'created_at' => $this->created_at,
            'updated_at' => $this->updated_at,
        ];
    }
}
```

GetMediaPlan.php

```php
<?php

namespace App\Http\Controllers\MediaPlan;

use App\Http\Controllers\Controller;
use App\Http\Resources\MediaPlanResource;
use App\Services\MediaPlanService;

class GetMediaPlan extends Controller
{
    public function __invoke(MediaPlanService $mpService)
    {
        return MediaPlanResource::collection(
            $mpService->get()
        );
    }
}
```

**Comment by staudenmeir** [comment]
Thanks, I can reproduce it now.

**Comment by staudenmeir** [comment]
This is caused by a change in `"nesbot/carbon":"^2.0"`. `return new Carbon;` also shows the difference.

You can (temporarily) fix this by adding `"nesbot/carbon": "^1.26.3"` to your `composer.json` file and running `composer update`.

**Comment by edbizarro** [comment]
Thanks!, with this temporarily fix I can have the old behavior back
